python/GafferRenderManUI/RenderManLightFilterUI.py

```diff
@@ -0,0 +1,66 @@
+##########################################################################
+#
+#  Copyright (c) 2025, Cinesite VFX Ltd. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import Gaffer
+import GafferRenderMan
+
+# \todo Refactor the GafferScene::LightFilter base class so this can be
+# registered there, and work for all subclasses. The main issue is that
+# there is no simple generic way of querying the required "ri:lightFilter:"
+# prefix from the subclass.
```

Comment:
Doesn't `lightFilter["__shader"]["type"].getValue()` get us what we want?

Reply:
It makes sense in ArnoldLightUI, because the Light base class doesn't know that all the derived classes are using an internal shader node, so it truly doesn't have an easy way of getting the type. But LightFilter knows about the internal node because it is the one that manages it, and can just use `["__shader"]["type"].getValue()`. Let's do that now rather add to the mess with another layer of metadata provision.

For 1.6 I intend to refactor the Light base class to use the same approach as LightFilter, and to do some further consolidation of metadata so there aren't so many disparate mechanisms for providing it.

Reply:
Ah ok, I didn't realize the light wasn't guaranteed to use an internal shader node, that makes sense now. I'll rework the light filter class so the todo-tentacles don't keep spreading.

Reply:
The latest push, https://github.com/GafferHQ/gaffer/pull/6351/commits/a277e0da0a2c6dc27cb2a30bd5f2c080e6149bce, moves the `userDefault` handling into `LightFilterUI`. In there I noted the fallback handling the `filter` suffix that can be removed when the light filter shader loading is reworked to omit that suffix.